sdk/typespec/src/http/status_code.rs

```diff
@@ -646,106 +665,134 @@ mod serde {
     }
 }
 
+impl Deref for StatusCode {
+    type Target = u16;
+
+    fn deref(&self) -> &Self::Target {
+        if let Self::UnknownValue(code) = self {
+            return code;
+        }
+
+        unsafe { std::mem::transmute_copy(&self) }
+    }
+}
+
 impl From<StatusCode> for u16 {
     fn from(code: StatusCode) -> u16 {
-        code as u16
+        match code {
+            StatusCode::UnknownValue(code) => code,
+            _ => *code,
+        }
     }
 }
 
-impl std::convert::TryFrom<u16> for StatusCode {
-    type Error = crate::Error;
-
-    fn try_from(num: u16) -> Result<Self, Self::Error> {
+impl From<u16> for StatusCode {
+    fn from(num: u16) -> Self {
         match num {
-            100 => Ok(StatusCode::Continue),
```

Comment:
Stupid question: Would it make sense to embody this in a table which has #, StatusCode, and String so that the two separate big match statements could be collapsed?

Reply:
If we had contiguous numbers that would be useful, but these are not contiguous. And it's incredibly risky to reference arbitrary references. WE could define consts, but that just ends up being even more work to maintain. Do you have a better suggestion?

Reply:
That's totally fair. And yeah, the compiler can (and will) do magic for non-contiguous match cases.